packages/build/src/npm-packages/push-tags.ts

```diff
@@ -0,0 +1,88 @@
+import type { SpawnSyncOptionsWithStringEncoding } from 'child_process';
+import {
+  EXCLUDE_RELEASE_PACKAGES,
+  MONGOSH_RELEASE_PACKAGES,
+  PROJECT_ROOT,
+} from './constants';
+import type { LernaPackageDescription } from './list';
+import { listNpmPackages as listNpmPackagesFn } from './list';
+import { spawnSync as spawnSyncFn } from '../helpers/spawn-sync';
+
+export function pushTags(
+  { useAuxiliaryPackagesOnly }: { useAuxiliaryPackagesOnly: boolean },
+  listNpmPackages: typeof listNpmPackagesFn = listNpmPackagesFn,
+  existsVersionTag: typeof existsTag = existsTag,
+  spawnSync: typeof spawnSyncFn = spawnSyncFn
+) {
+  const allReleasablePackages = listNpmPackages().filter(
+    (packageConfig) => !EXCLUDE_RELEASE_PACKAGES.includes(packageConfig.name)
+  );
+
+  const packages: LernaPackageDescription[] = useAuxiliaryPackagesOnly
+    ? allReleasablePackages.filter(
+        (packageConfig) =>
+          !MONGOSH_RELEASE_PACKAGES.includes(packageConfig.name)
+      )
+    : allReleasablePackages;
+
+  const commandOptions: SpawnSyncOptionsWithStringEncoding = {
+    stdio: 'inherit',
+    cwd: PROJECT_ROOT,
+    encoding: 'utf8',
+    env: {
+      ...process.env,
+    },
+  };
+
+  for (const packageInfo of packages) {
+    const { name, version } = packageInfo;
+    const newTag = `${name}@${version}`;
+
+    if (existsVersionTag(newTag)) {
+      console.warn(`${newTag} tag already exists. Skipping...`);
+      continue;
+    }
+    spawnSync('git', ['tag', '-a', newTag, '-m', newTag], commandOptions);
+  }
+
+  if (!useAuxiliaryPackagesOnly) {
+    const mongoshVersion = packages.find(
+      (packageConfig) => packageConfig.name === 'mongosh'
+    )?.version;
+
+    if (!mongoshVersion) {
+      throw new Error('mongosh package not found');
+    }
+
+    const newVersionTag = `v${mongoshVersion}`;
+
+    if (!existsVersionTag(newVersionTag)) {
+      console.info(`Creating v${mongoshVersion} tag...`);
+      spawnSync(
+        'git',
+        ['tag', '-a', newVersionTag, '-m', newVersionTag],
+        commandOptions
+      );
+    } else {
+      console.warn(`${newVersionTag} tag already exists. Skipping...`);
+    }
+  }
+
+  spawnSync('git', ['push', '--follow-tags'], commandOptions);
+}
+
+/** Returns true if the tag exists in the remote repository. */
+export function existsTag(tag: string): boolean {
```

Comment:
I added this as another sanity check though maybe it's a bit of an overkill / slow when done for every auxiliary package.

Reply:
I think this is fine 👍